src/System.Text.JsonLab.Serialization/System/Text/Json/Serialization/JsonConverter.Sync.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Collections.Generic;
+
+namespace System.Text.Json.Serialization
+{
+    public static partial class JsonConverter
+    {
+        public static T FromJson<T>(string json, JsonConverterSettings options = null)
+        {
+            if (json == null)
+                throw new ArgumentNullException(nameof(json));
+
+            return (T)FromJson(json, typeof(T), options);
+        }
+
+        public static object FromJson(string json, Type returnType, JsonConverterSettings options = null)
+        {
+            if (json == null)
+                throw new ArgumentNullException(nameof(json));
+
+            if (returnType == null)
+                throw new ArgumentNullException(nameof(returnType));
+
+            return FromJsonInternal(json, returnType, options);
+        }
+
+        private static object FromJsonInternal(string json, Type returnType, JsonConverterSettings options = null)
+        {
+            if (options == null)
+                options = s_DefaultSettings;
+
+            byte[] jsonBytes = s_utf8Encoding.GetBytes(json);
```

Comment:
Can we instead use span-based transcoding APIs and write the UTF-8 bytes to a stackalloc span for json.Length <= 512 and borrow from the array pool for > 512, so we can avoid these intermediary allocations?

We can return the array back to the pool (and clear it) before returning the result.

Reply:
I'll add a perf todo for now. How important do you think string-based scenarios are?

I'll be sure to add some API doc that the UTF8 approach is preferred for performance - meaning be sure to specify UTF8 content encoding.

Reply:
Since usability burden on the serializer/deserializer is higher than the jsonreader/writer (and used a lot more), I would imagine people to use what they are familiar with (i.e. strings). If we can make it faster for them as well, that would be awesome.